### PR TITLE
feat(PE-6612): update manage asset table

### PR DESCRIPTION
**Pull request by atticusofsparta**

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -36,6 +36,7 @@
     "@typescript-eslint/no-unused-vars": ["error", {}],
     "@typescript-eslint/no-explicit-any": "off",
     "no-useless-escape": "off",
-    "react/no-children-prop": "off"
+    "react/no-children-prop": "off",
+    "react/prop-types": "off"
```
Comment (kunstmusik) [inline .eslintrc:40]: Why is react/prop-types off here? Seems like it'll hide problems that should be fixed.

File: src/utils/searchUtils/searchUtils.ts [hunk 2]
```diff
@@ -157,3 +157,23 @@ export function doAntsRequireUpdate({
 
   return getAntsRequiringUpdate({ ants, luaSourceTx }).length > 0;
 }
+
+export function camelToReadable(camel: string) {
```
Comment (kunstmusik) [inline src/utils/searchUtils/searchUtils.ts:161]: Would be good to have a unit test for this

File: src/utils/searchUtils/searchUtils.ts [hunk 3]
```diff
@@ -157,3 +157,23 @@ export function doAntsRequireUpdate({
...
+  const words = camel.replace(/([A-Z])/g, ' $1').toLowerCase();
+  return words.charAt(0).toUpperCase() + words.slice(1);
+}
+
+export function getOwnershipStatus(
```
Comment (kunstmusik) [inline src/utils/searchUtils/searchUtils.ts:166]: prefer a single expression of

```
return (owner === walletAddress) ? 
         'owner' : 
         (walletAddress && controllers.includes(walletAddress)) ? 
         'controller' : 
         undefined;
```

File: src/utils/common/common.tsx [hunk 4]
```diff
@@ -10,6 +10,23 @@ export function formatDate(epochMs: number): string {
   return new Date(epochMs).toISOString().split('T')[0];
 }
 
+export function formatVerboseDate(timestamp: number | string): string {
```
Comment (kunstmusik) [inline src/utils/common/common.tsx:13]: would be good to have a comment here with an example of the expected date string output; also a unit test would be good

File: src/state/contexts/ModalState.tsx [hunk 5]
```diff
@@ -5,6 +5,7 @@ import { ModalAction } from '../reducers/ModalReducer';
 
 export type ModalState = {
   showUpgradeAntModal: boolean;
+  showEditUndernameModal: boolean;
```
Comment (kunstmusik) [inline src/state/contexts/ModalState.tsx:8]: Not for this PR, but we should discuss the way modal's are done in Network Portal and see about a similar pattern for this app in the future.

File: src/components/pages/Manage/Manage.tsx [hunk 6]
```diff
@@ -1,55 +1,32 @@
...
-  }, [path]);
-
-  function updatePage(page: number) {
-    setTablePage(page);
-  }
+  const [search, setSearch] = useState<string>('');
 
   return (
-    <div className="page" style={{ overflow: 'auto' }}>
+    <div className="page" style={{ overflow: 'auto', paddingTop: '0px' }}>
```
Comment (kunstmusik) [inline src/components/pages/Manage/Manage.tsx:29]: maybe go ahead and switch this to tailwind and use `className="page pt-0 overflow-auto"`

File: src/components/layout/Layout/Layout.tsx [hunk 7]
```diff
@@ -38,11 +41,25 @@ function Layout() {
       >
         <NavBar />
       </div>
+
+      <Progress
```
Comment (kunstmusik) [inline src/components/layout/Layout/Layout.tsx:45]: This has a strange effect when load or reload completes, the bar animates from full all the way back down to 0. It's a bit jarring, IMO, and the bar should just disappear.

File: src/components/data-display/tables/DomainsTable.tsx [hunk 8]
```diff
@@ -0,0 +1,467 @@
...
+        };
+        newTableData.push(data);
+      });
+
+      setTableData(newTableData);
+    }
+  }, [domainData, loading]);
+
+  useEffect(() => {
+    if (filter) {
```
Comment (kunstmusik) [inline src/components/data-display/tables/DomainsTable.tsx:145]: I think there may be something off here. If I put in a text that triggers the filter, then hit backspace repeatedly to clear it out, the filter updates and clears out. If instead I select the text and delete it or select and cut the text with cmd-x, the table does not update.

File: .eslintrc (hunk 1 above)
Comment (atticusofsparta) [reply .eslintrc:40]: It was requiring prop types in implementations as well as definitions. Typescript already covers this.

**Review by kunstmusik — APPROVED** [review]

**Pull request merged**